src/App.tsx

```diff
@@ -1,16 +1,195 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import {
+  getTodos,
+  createTodo,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoErrorNotification } from './components/TodoErrorNotification';
+import { TodoFooter } from './components/TodoFooter';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { SortType } from './types/SortType';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [sortType, setSortType] = useState(SortType.ALL);
+  const [isAdding, setIsAdding] = useState(false);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [title, setTitle] = useState('');
+  const [isLoader, setIsLoader] = useState(false);
+  const [isLoaderCompletedTodo, setIsLoaderCompletedTodo] = useState(false);
+  const [isLoaderUnCompletedTodo, setIsLoaderUnCompletedTodo] = useState(false);
+  const [isDoubleClick, setIsDoubleClick] = useState(false);
+  const [changedTodoTitle, setChangedTodoTitle] = useState('');
+
+  const completedTodosIds = [...todos].filter(todo => todo.completed)
+    .map(todo => todo.id);
+  const unCompletedTodosIds = [...todos].filter(todo => !todo.completed)
+    .map(todo => todo.id);
+
+  const isAllTodosCompleted = [...todos].every(todo => todo.completed);
+
+  const getFilteredTodos = () => {
+    let filteredTodos = [...todos];
+
+    switch (sortType) {
+      case SortType.ACTIVE:
+        filteredTodos = filteredTodos.filter(todo => !todo.completed);
+        break;
+      case SortType.COMPLETED:
+        filteredTodos = filteredTodos.filter(todo => todo.completed);
+        break;
+
+      default:
+    }
+
+    return filteredTodos;
+  };
+
+  const handlerSubmitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!newTodoTitle) {
+      setError('Title can\'t be empty');
+    } else {
```

Comment:
```suggestion
    if (!newTodoTitle) {
      setError('Title can\'t be empty');
      
      return
     }
```
it's better for readability